frontend/src/components/pages/TranslationPage.tsx

```diff
@@ -25,8 +25,10 @@ type Content = {
 };
 
 type HistoryStack = {
-  Undo: Array<{ lineIndex: number; content: string }>;
-  Redo: Array<{ lineIndex: number; content: string }>;
+  Undo: Array<StoryLine[]>;
```

Comment:
having a somewhat hard time following the logic -  why switch to an array of `StoryLine[]`?

Reply:
What Emma said is exactly correct,  we can't actually know which line the user will change until they actually change it, so when the state is stored, it needs to be all lines.